src/service/fixed-layer.js

```diff
@@ -557,9 +536,18 @@ export class FixedLayer {
 
     // Update `top`. This is necessary to adjust position to the viewer's
     // paddingTop.
-    if (state.top && (state.fixed || (state.sticky && !this.transfer_))) {
-      setStyle(element, 'top', `calc(${state.top} + ${this.paddingTop_}px)`);
+    // TODO
+    if (state.top) {
+      if (state.fixed || !this.transfer_) {
+        setStyle(element, 'top', `calc(${state.top} + ${this.paddingTop_}px)`);
+      } else if (this.committedPaddingTop_ === this.paddingTop_) {
+        setStyle(element, 'top', state.top);
```

Comment:
Ditto: why do we need to update `top` styles of non-fixed/non-sticky elements? Same is with `else` below?